### PR TITLE
Refine CRD deprecations

### PR DESCRIPTION
**Pull request by teozkr**
## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)

### CODE REVIEW

File: src/crd.rs [hunk 1]
```diff
@@ -69,6 +69,11 @@ impl<T> CustomResourceExt for T where T: kube::CustomResourceExt {}
 /// - `client`: Client to connect to Kubernetes API and create the CRD with
 /// - `timeout`: If specified, retries creating the CRD for given `Duration`. If not specified,
 ///     retries indefinitely.
+#[deprecated(
```
Comment (soenkeliebau) [inline src/crd.rs:72]: I think we actually have a ticket somewhere https://github.com/stackabletech/issues/issues/73  about this and wanted to make this configurable in the future..

**Review by soenkeliebau — COMMENTED** [review]
Just one question..

File: src/crd.rs (hunk 1 above)
Comment (teozkr) [reply src/crd.rs:72]: Wasn't that pre-k8s? That made some sense when we wanted to say "ok, install this Debian package to install and we'll install the k8s resources we need", less so to have to say "install these k8s resources and we'll create this final one for you".

Comment (soenkeliebau) [reply src/crd.rs:72]: that is true I guess.. but wouldn't it still be an option to just start an operator with `kubectl run ...` if you dislike or don't have Helm?

Comment (teozkr) [reply src/crd.rs:72]: You'd still need to do a bunch of setup in the background, just to ensure that the correct RBAC permissions are applied. The default ServiceAccount certainly won't have permission to create CRD objects in most clusters.

The "correct" solution there would be to `kubectl apply` the kustomize manifests (https://github.com/stackabletech/issues/issues/46).

Comment (soenkeliebau) [reply src/crd.rs:72]: ok, I'll shut up :)

**Review by soenkeliebau — APPROVED** [review]

**Pull request merged**